src/rgw/rgw_rados.cc

```diff
@@ -6576,14 +6584,14 @@ int RGWRados::put_system_obj_data(void *ctx, rgw_obj& obj, bufferlist& bl,
 
   ObjectWriteOperation op;
 
-  if (exclusive)
+  if (exclusive) {
     op.create(true);
-
-  if (ofs == -1) {
-    op.write_full(bl);
-  } else {
-    op.write(ofs, bl);
   }
+
+  op.set_alloc_hint2(bl.length(), bl.length(), ALLOC_HINT_FLAG_SEQUENTIAL_WRITE |
+                                               ALLOC_HINT_FLAG_SEQUENTIAL_READ |
+                                               ALLOC_HINT_FLAG_APPEND_ONLY);
```

Comment:
It's harmless on an existing object; ideally it should only be called when the object is first created though (just for efficiency).